### PR TITLE
Builtby Variable

### PR DESCRIPTION
**Pull request by Super3222**
This will add a "builtby" variable, which will allow admins to see who exactly was associated with creating objects, structures, machinery, etc. It will also differentiate between those loaded with the map and those created in game.

I'll do the same for craftable items/weapons after ghetto armoury gets merged.

How to get to it: View Variables > Type next to "Search:" builtby > It displays the users ckey and the users current in game name.
#25 requested this.

### CODE REVIEW

File: code/game/machinery/machinery.dm [hunk 1]
```diff
@@ -124,6 +124,12 @@ Class Procs:
 	machines += src
 	SSmachine.processing += src
 	power_change()
+	var/mob/user = usr
+	if(ticker.current_state == GAME_STATE_PLAYING)
+		src.builtby = "This was created by [user.ckey]/[user.name]."
```
Comment (Cruix) [inline code/game/machinery/machinery.dm:129]: You should check to make sure the user isn't null, just in case.

**Comment by ghost** [comment]
OH THANK GOD

**Comment by Alek2ander** [comment]
Why use a separate variable when you can put this info in fingerprints?

**Comment by Super3222** [comment]
Eh, I might close this for two reasons.
1. Fingerprints basically do the same thing. We really don't need this.
2. Some things in the code may show up as null unexpectedly after construction, and there's no way around it.

**Pull request closed without merging**